File: client-app/src/app/models/evento.ts
import { Profile } from "./profile";

export interface Evento {
    id: string;
    title: string;
    url: string;
    startDate: Date | null;
    endDate: Date | null;
    description: string;
    category: string;
    city: string;
    venue: string;
    hostUsername: string;
    isCancelled: boolean;
    isGoing: boolean;
    isHost: boolean;
    host?:Profile;
    appUserId: string;
    asistentes?: Profile[];
}

export class Evento implements Evento {
    constructor(init?: EventoFormValues) {
        Object.assign(this, init);
    }
}

export class EventoFormValues {
    id?: string = undefined;
    url: string ='';
    title: string = '';
    category: string ='';
    description: string = '';
    startDate: Date | null = null;
    endDate: Date | null = null;
    city: string = '';
    venue: string = '';
    appUserId: string = '';
    constructor(evento?: EventoFormValues) {
        if (evento) {
            this.id = evento.id;
            this.url = evento.url;
            this.title = evento.title;
            this.description = evento.description;
            this.category = evento.category;
            this.startDate = evento.startDate;
            this.endDate = evento.endDate;
            this.venue = evento.venue;
            this.city = evento.city;
            this.appUserId = evento.appUserId;
        }
    }
}

File: Domain/Evento.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Evento
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Url { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public string City { get; set; }
        public string Venue { get; set; }
        public string AppUserId {get; set;}
        public AppUser AppUser {get; set;}
        public bool IsCancelled {get; set;}
        
        public ICollection<EventoAsistente> Asistentes { get; set; } = new List<EventoAsistente>();
        public ICollection<GalleryEvento> GalleryEventos { get; set; } = new List<GalleryEvento>();
        public ICollection<Gallery> Galleries { get; set; } = new List<Gallery>();
        
    }
}